src/Engine/Component/PlayerComponents/PlayerControllerComponent.cpp

```diff
@@ -0,0 +1,93 @@
+#include "PlayerControllerComponent.hpp"
+
+#include "IO/Window.hpp"
+#include "IO/Mouse.hpp"
+#include "IO/Keyboard.hpp"
+#include "Component/SpatialComponents/SpatialComponent.hpp"
+#include "Component/CameraComponents/CameraComponent.hpp"
+#include "Component/SpatialComponents/PhysicsComponents.hpp"
+#include "Component/PostCollisionComponents/GroundComponent.hpp"
+#include "System/SpatialSystem.hpp"
+
+
+
+PlayerControllerComponent::PlayerControllerComponent(float lookSpeed, float moveSpeed, float jumpSpeed) :
```

Comment:
Looks like a lot of functionality here is extremely similar to the regular camera controller component. Could be worth creating a generic camera controller component class and deriving it?